### ISSUE TITLE
Test if these are possible bugs in the editor

### ISSUE THREAD

**Issue by lilimelgar** [CLARIAH/vocab-registry-editor#5]
When editing an existing record using the latest version of the editor, these two issues occur:

1) the edited record is saved with a new file name, e.g., the original record file name was "record-499", once it's opened with the editor and saved with new content, the original version is saved with a new file name: "record-499.2024-05-07.xml", where 2024-05-07 was the creation date of the original file. There is a new file created with the name "record-499" which contains the most recent (edited) version. This results in having then two files with different versions for this record. Not sure if this is the intended behavior.

2) The new version of the edited file, i.e., "record-499" doesn't preserve the hierarchical structure of the original record in XML, it's only seen as flat text (see example: https://knaw-my.sharepoint.com/:u:/g/personal/liliana_melgar_di_huc_knaw_nl/EeKQKmsCo9VKlLSuLIIALr8BivIyJN5syVj64of57ZjLkw?e=qtrW4q).

**Comment by menzowindhouwer** [comment]
The results are as expected:
1. the editing history is preserved,  in this case record-499.xml always refers to the latest version;
2. its a proper XML record it just misses extra whitespace to make the structure more human readable.

To add the whitespace for human consumption load the record  in an XML aware editor like oXygen or use ``xmllint --format record-499.xml`` on the command line.

**Comment by menzowindhouwer** [comment]
This is not a bug, the results are as expected:
1. the editing history is preserved,  in this case record-499.xml always refers to the latest version;
2. its a proper XML record it just misses extra whitespace to make the structure more human readable.The record will function fine within the system and without information loss.

To add the whitespace for human consumption load the record  in an XML aware editor like oXygen and use _Format and indent_ or use ``xmllint --format record-499.xml`` on the command line.

**Comment by menzowindhouwer** [comment]
This is not a bug, the results are as expected:
1. the editing history is preserved,  in this case record-499.xml always refers to the latest version;
2. its a proper XML record it just misses extra whitespace to make the structure more human readable.The record will function fine within the system and without information loss.

To add the whitespace for human consumption load the record  in an XML aware editor like oXygen and use _Format and indent_ or use ``xmllint --format record-499.xml`` on the command line.

**Comment by lilimelgar** [comment]
Thanks @menzowindhouwer for the useful clarifications. Here some follow-up comments/observations:

1. Good that the history/versioning is preserved. It will be important then to store the creation date as a separate field and store the editing/modification date as a new field, allowing to configure access to both dates, or at least one of them, from the editor's interface since it displays the date in which the record was updated:
![Screenshot 2024-12-30 at 14 17 23](https://github.com/user-attachments/assets/8e1d676d-908c-440c-a339-beee6d0e5d5f). This record was created on 2024-06-12, but this date is overwritten in the new version.

2. I have tested in different editors and on the browser, and it is quite difficult/tricky to make it human readable. In Oxygen it works quite well, but this tool is paid. I think for the end user it would be better, if possible, if the content would be generated in a ready to use human readable version (hierarchical).
